src/Domain/Tests/DomainCollectionTest.php

```diff
@@ -124,9 +127,11 @@ public function testLazySlice(): void
         self::assertSame([1 => 2], iterator_to_array(($collection = static::createLazyCollection([1, 2, 3, null, 5], $visited))->slice(1, 1)));
         self::assertSame([1, 2, 3], $visited);
 
-        $this->assertUnrewindableGenerator();
+        $result = static::createLazyCollection([1, 2, 3])->slice(0);
```

Comment:
i suggest to reuse the previous $collection, e.g. `$result = $collection->slice(0)`  so we test a second call doesnt break anymore, which it did before.